docs/utilities/recipes/ingredients.md

```diff
@@ -0,0 +1,173 @@
+Ingredients
+===========
+
+`Ingredient`s are predicate handlers for item-based inputs which check whether a certain `ItemStack` meets the condition to be a valid input in a recipe. All [vanilla recipes][recipes] that take inputs use an `Ingredient` or a list of `Ingredient`s, which is then merged into a single `Ingredient`.
+
+Custom Ingredients
+------------------
+
+Custom ingredients can be specified by setting `type` to the name of the [ingredient's serializer][serializer] with only [one exception][compound]. When no type is specified, `type` defaults to the vanilla ingredient `minecraft:item`. Custom ingredients can also easily be used in [data generation][datagen].
+
+### Forge Types
+
+Forge provides a few additional `Ingredient` types for programmers to implement. 
+
+#### CompoundIngredient
+
+Though they are functionally identical, Compound ingredients replaces the way one would implement a list of ingredients would in a recipe. They work as a set OR where the passed in stack must be within at least one of the supplied ingredients. This change was made to allow custom ingredients to work correctly within lists. As such, **NO TYPE** needs to be specified.
+
+```json5
+// For some input
+[
+  // At least one of these ingredients must return true to succeed
+  {
+    // Ingredient
+  },
+  {
+    // Custom ingredient
+    "type": "examplemod:example_ingredient"
+  }
+]
+```
+
+#### NBTIngredient
+
+`NBTIngredient`s compare the item, damage, and the share tags (as defined by `IForgeItem#getShareTag`) on an `ItemStack` for equivalency. This can be used by specifying the `type` as `forge:nbt`.
+
+```json5
+// For some input
+{
+  "type": "forge:nbt",
+  "item": "examplemod:example_item",
+  "nbt": {
+    // Add nbt data (must match exactly what is on the stack)
+  }
+}
+```
+
+### PartialNBTIngredient
+
+`PartialNBTIngredient`s are a looser version of [`NBTIngredient`][nbt] as they compare against a single or set of items and only keys specified within the tag. This can be used by specifying the `type` as `forge:partial_nbt`.
```

Comment:
Worth mentioning this also compares the share tag, ensures its consistent on client and server